### PR TITLE
FEAT: Add tol option to best response related methods

### LINKED ISSUE #21: Add tol option to best response related methods
See https://github.com/QuantEcon/QuantEcon.py/pull/273/commits/f98ed11306dbf34049decb73ddfe3115d2735ff6.

### PR DESCRIPTION
**Pull request by QBatista**
Close #21 #75

### CODE REVIEW

**Comment by oyamad** [comment]
Thanks @QBatista 

* Will you also fix the bug #73?
* Add test for `pure_nash`; see https://github.com/QuantEcon/QuantEcon.py/commit/ec1512189d64868cd376e6ab62efc9feb0d068d6#diff-f7890964c2217333ac4cdb3bf234c2c1.

File: src/normal_form_game.jl [hunk 1]
```diff
@@ -306,7 +306,7 @@ end
 """
     best_response(player, opponents_actions, payoff_perturbation)
 
-Return the perturbed best response to `opponents_actions`.
+Return the perturbed best responses to `opponents_actions`.
```
Comment (oyamad) [inline src/normal_form_game.jl]: Shouldn't this be singular?

File: src/normal_form_game.jl [hunk 2]
```diff
@@ -315,21 +315,36 @@ Return the perturbed best response to `opponents_actions`.
...
+      return indmax(payoffs)
+    elseif tie_breaking == "random"
+        payoff_max = maximum(payoffs)
+        brs = find(x -> x >= payoff_max - tol, payoffs)
+        return rand(brs)
+    else
+        throw(ArgumentError(
+            "tie_breaking must be one of 'smallest' or 'random'"
+        ))
+    end
```
[the remark this answers is not in the document]
Comment (QBatista) [reply src/normal_form_game.jl]: I did this to match the Python version which has both. Do you want me to get rid of it?

[the remark this answers is not in the document]
Comment (oyamad) [reply src/normal_form_game.jl]: Yes please.

File: src/normal_form_game.jl [hunk 3]
```diff
@@ -586,21 +589,25 @@ Return true if `action_profile` is a Nash equilibrium.
...
 - `action::Action` : Integer (pure action) or vector of reals (mixed action).
+- `tol::Float64` : Tolerance to be used to determine best response actions.
 
 # Returns
 
 - `::Bool`
 """
-is_nash(g::NormalFormGame{1}, action::Action) =
-    is_best_response(g.players[1], action, nothing)
+is_nash(g::NormalFormGame{1}, action::Action, tol::Float64=1e-8) =
```
Comment (oyamad) [inline src/normal_form_game.jl]: `tol` must be a keyword argument.

File: src/normal_form_game.jl [hunk 4]
```diff
@@ -586,21 +589,25 @@ Return true if `action_profile` is a Nash equilibrium.
...
+    is_best_response(g.players[1], action, nothing, tol=tol)
+
+is_nash(g::NormalFormGame{1}, action_profile::ActionProfile) =
```
Comment (oyamad) [inline src/normal_form_game.jl]: Add `tol` argument.

**Comment by QBatista** [comment]
@oyamad This should be ready -- do you want me to squash into a few logical commits before reviewing it?

**Comment by oyamad** [comment]
Yes, please.

**Pull request merged**